src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpContext.cs

```diff
@@ -165,6 +158,30 @@ protected void InitializeContext()
             _bodyOutput = new OutputProducer(pipe);
         }
 
+        private unsafe void ModifyFeatureCollection()
+        {
+            ResetFeatureCollection();
+
+            if (_options.ForwardClientCertificate)
+            {
+                // Only set the ITlsConnectionFeature if there is a cert available.
+                if (NativeRequest->pSslInfo != null &&
+                    NativeRequest->pSslInfo->pClientCertInfo != null &&
+                    NativeRequest->pSslInfo->pClientCertInfo->pCertEncoded != null &&
+                    NativeRequest->pSslInfo->pClientCertInfo->CertEncodedSize != 0)
+                {
+                    _currentITlsConnectionFeature = new IISTlsConnectionFeature(
```

Comment:
Why change the pattern and not implement the feature on context?

Reply:
We need to check that the cert exists before adding the feature. This matches the out of proc behavior.

Reply:
We do this for websockets and null-out the feature if it's disabled.